### PR TITLE
Ports Not-Uno and changes card stacks

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Ports https://github.com/BeeStation/BeeStation-Hornet/pull/1422

## Why It's Good For The Game
https://www.youtube.com/watch?v=CapLbFlOVOs

## Changelog
:cl:
add: New "Kotahi" card game now available in fun vendors
add: Cards stacks will now display the top 3 cards in your hand instead of a static image
/:cl:

### CODE REVIEW

**Comment by trollbreeder** [comment]
Speedmerge now, omega based

**Comment by mrhugo13** [comment]
**### YOU HAVE UNO!!!!!!!!!!!!!!!!!!!!!!!!!!!!!**

**Comment by spookydonut** [comment]
why not white cards with color shifting? for 1-9 RS

**Comment by ExcessiveUseOfCobblestone** [comment]
is.... is this allowed?

**Comment by Tlaltecuhtli** [comment]
i searched on google and it doesnt seem copyrighted

**Comment by TiviPlus** [comment]
Theres a copyright on the cards(You cant just copypaste the card pictures) and a trademark on the way (designwise) the name is written in conjunction with sold video games, so yes, we can "do this".

Also I have no clue what color shifting is but I'll look into it

**Comment by Crossedfall** [comment]
Copyright isn't the issue, it's the trademark. 

https://trademarks.justia.com/730/15/uno-73015277.html

I've asked the author of our PR to use a generic name.

**Comment by ExcessiveUseOfCobblestone** [comment]
or we could just call it something else and avoid the issue entirely :)

**Comment by optimumtact** [comment]
@cacogen easy for you to say when you bear none of the risk

change the name to kotahi

**Comment by Dingo-Dongler** [comment]
psst, Beestation/Beestation-Hornet#1457 was also made to make this work WAY better. Also I'll change the name soon.

**Comment by cacogen** [comment]
juan uno sending a mexican cartel hitsquad to mso's house for trademark infringement

**Review by Dingo-Dongler — COMMENTED** [review]
Did you test the -= choice thing? It didn't work for my neck deck with duplicates initially.

**Comment by JJRcop** [comment]
Because y'all acknowledge the potential for legal issues, that destroys plausible deniability. It must be changes

**Comment by TiviPlus** [comment]
Just to be safe I'll go ahead and change the name then.

>Did you test the -= choice thing? It didn't work for my neck deck with duplicates initially.

"There are 108 cards in a Uno deck. There are four suits, Red, Green, Yellow and Blue, each consisting of one 0 card, two 1 cards, two 2s, 3s, 4s, 5s, 6s, 7s, 8s and 9s; two Draw Two cards; two Skip cards; and two Reverse cards. In addition there are four Wild cards and four Wild Draw Four cards." I have no clue what you mean but if you take every card out it conforms to this

Yea sure why not I'll port whatever it is you linked too, might as well

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -958,6 +948,23 @@
...
+
+	var/k
+	if(overlay_cards == 2)
+		k = 1
+	else
+		k = overlay_cards - 2
+
+	for(var/i = k; i <= overlay_cards; i++)
+		var/card_overlay = image(icon=src.icon,icon_state="sc_[currenthand[i]]_[deckstyle]",pixel_x=(1-i+k)*3,pixel_y=(1-i+k)*3)
+		add_overlay(card_overlay)
```
Comment (bobbahbrown) [inline code/game/objects/items/toys.dm:966]:
```suggestion
	var/k = overlay_cards == 2 ? 1 : overlay_cards - 2
	for(var/i = k; i <= overlay_cards; i++)
		var/card_overlay = image(icon=src.icon,icon_state="sc_[currenthand[i]]_[deckstyle]",pixel_x=(1-i+k)*3,pixel_y=(1-i+k)*3)
		add_overlay(card_overlay)
```

**Review by bobbah — CHANGES_REQUESTED** [review]

File: code/game/objects/items/toys.dm (hunk 1 above)
Comment (bobbahbrown) [reply code/game/objects/items/toys.dm:966]: if you are just trying to avoid k being less than 1 maybe just use max or clamp

Comment (TiviPlus) [reply code/game/objects/items/toys.dm:966]: @Dingo-Dongler This ones for you too

**Pull request merged**